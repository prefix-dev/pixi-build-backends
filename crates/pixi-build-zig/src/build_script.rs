use minijinja::Environment;
use serde::Serialize;

#[derive(Serialize)]
pub struct BuildScriptContext {
    /// Any additional args to pass to `zig build`
    pub extra_args: Vec<String>,

    /// The platform that is running the build.
    pub is_bash: bool,
}

impl BuildScriptContext {
    pub fn render(&self) -> String {
        let env = Environment::new();
        let template = env
            .template_from_str(include_str!("build_script.j2"))
            .unwrap();
        template.render(self).unwrap().trim().to_string()
    }
}

#[cfg(test)]
mod test {
    use rstest::*;

    #[rstest]
    fn test_build_script(#[values(true, false)] is_bash: bool) {
        let context = super::BuildScriptContext {
            extra_args: vec![],
            is_bash,
        };
        let script = context.render();

        let mut settings = insta::Settings::clone_current();
        settings.set_snapshot_suffix(if is_bash { "bash" } else { "cmdexe" });
        settings.bind(|| {
            insta::assert_snapshot!(script);
        });
    }

    #[rstest]
    fn test_build_script_with_extra_args(#[values(true, false)] is_bash: bool) {
        let context = super::BuildScriptContext {
            extra_args: vec!["-Doptimize=ReleaseFast".to_string()],
            is_bash,
        };
        let script = context.render();

        let mut settings = insta::Settings::clone_current();
        settings.set_snapshot_suffix(if is_bash { "bash" } else { "cmdexe" });
        settings.bind(|| {
            insta::assert_snapshot!(script);
        });
    }
}
