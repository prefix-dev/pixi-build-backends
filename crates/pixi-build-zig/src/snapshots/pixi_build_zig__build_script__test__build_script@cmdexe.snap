---
source: crates/pixi-build-zig/src/build_script.rs
expression: script
---
zig build --prefix %LIBRARY_PREFIX%

if errorlevel 1 exit 1
